tools/oboe-main.jl

```diff
@@ -73,38 +93,53 @@ function processOboe(name, agg; fips=fipsAll, useNW=false, force=false)
     d = Oboe.mkAP_pop_dict(ns) 
     #go on, compute the nodes' passenger shares (add the :shr col), with `d` in mind
     ns2 = Oboe.assignPsgShares(ns,d)
-    ns2 |> myshow
+    #ns2 |> myshow
 
     #generate matrix of AP-to-AP flights with aux. arrays mapping indices to IATA codes
     fmx = Oboe.mkFlightMx2(pBTS, ns2.IATA_Code |> unique; daily=true)
     #generate commute table
     cmt = Oboe.rdTidyWfsByFIPS(fips, ns2)
-    cmt |> myshow
+    #cmt |> myshow
 
     #do the processing, aggregating by tract/state/county/ap as selected in the args
-    skipagg = false #this will be true if by-tract aggregation is requested
-                    #and thus no additional steps are needed
-    if agg == "tra"
-        skipagg = true
-    elseif agg == "cty"
-        aggregated = Oboe.aggByCty(ns2)
-        @time partitioned = Oboe.partByCty(ns2, aggregated)
-    elseif agg == "ap"
-        aggregated = Oboe.aggByAP(ns2)
-        @time partitioned = Oboe.partByAP(ns2, aggregated)
-    elseif agg == "ste"
-        aggregated = Oboe.aggBySte(ns2)
-        @time partitioned = Oboe.partBySte(ns2, aggregated)
-    end
+    
+
+    println("Processing started. Aggregation mode [$agg] selected.")
+    aggnpart = dispatchAggPart(agg) #dispatch the aggregate/partition function pair
+    println("Aggregating")
+    @time aggregated = ns2 |> aggnpart[1] #aggregate as selected by dispatcher
+    println("Partitioning")
+    @time partitioned = aggnpart[2](ns2,aggregated) #partition as selected by dispatcher
 
+    skipagg = (agg == "tra") ? true : false #no aggregation necessary if "tra" requested
```

Comment:
That ternary expression seems redundant

Reply:
Could you please elaborate? I retained `skipagg` to distpatch between 2- and 4-argument `mk[Cmt|Psg]Mx`, and I find it more clear than a direct `agg == "tra"` check.